v2/managed-io-to-managed-io/src/main/java/com/google/cloud/teleport/v2/templates/ManagedIOToManagedIO.java

```diff
@@ -0,0 +1,264 @@
+import org.apache.beam.sdk.managed.Managed;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.sdk.values.Row;
+
+/**
+ * The {@link ManagedIOToManagedIO} pipeline is a flexible template that can read from any Managed
+ * I/O source and write to any Managed I/O sink. This template supports all available Managed I/O
+ * connectors including ICEBERG, ICEBERG_CDC, KAFKA, and BIGQUERY.
+ *
+ * <p><b>Pipeline Requirements</b>
+ *
+ * <ul>
+ *   <li>The source and sink configurations must be valid for the specified connector types.
+ *   <li>Required permissions for accessing the source and sink systems.
+ * </ul>
+ *
+ * <p>Check out <a
+ * href="https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/managed-io-to-managed-io/README.md">README</a>
+ * for instructions on how to use or modify this template.
+ */
+@Template(
+    name = "Managed_IO_to_Managed_IO",
+    category = TemplateCategory.BATCH,
+    displayName = "Managed I/O to Managed I/O",
+    description =
+        "The Managed I/O to Managed I/O template is a flexible pipeline that can read from any "
+            + "Managed I/O source and write to any Managed I/O sink. This template supports all "
+            + "available Managed I/O connectors including ICEBERG, ICEBERG_CDC, KAFKA, and BIGQUERY. "
+            + "The template uses Apache Beam's Managed API to provide a unified interface for "
+            + "configuring different I/O connectors through simple configuration maps.",
+    optionsClass = Options.class,
+    flexContainerName = "managed-io-to-managed-io",
+    documentation = "https://cloud.google.com/dataflow/docs/guides/managed-io",
+    contactInformation = "https://cloud.google.com/support",
+    requirements = {
+      "The source and sink configurations must be valid for the specified connector types.",
+      "Required permissions for accessing the source and sink systems."
+    },
+    streaming = false)
+public class ManagedIOToManagedIO {
+
+  /**
+   * The {@link Options} class provides the custom execution options passed by the executor at the
+   * command-line.
+   */
+  public interface Options extends PipelineOptions {
+
+    @TemplateParameter.Text(
+        order = 1,
+        description = "Source Managed I/O connector type",
+        helpText =
+            "The type of Managed I/O connector to use as source. "
+                + "Supported values: ICEBERG, ICEBERG_CDC, KAFKA, BIGQUERY.")
+    String getSourceConnectorType();
+
+    void setSourceConnectorType(String value);
+
+    @TemplateParameter.Text(
+        order = 2,
+        description = "Source configuration (JSON)",
+        helpText =
+            "JSON configuration for the source Managed I/O connector. "
+                + "The configuration format depends on the connector type. "
```

Comment:
Could we link to https://cloud.google.com/dataflow/docs/guides/managed-io#supported for this?

@chamikaramj @ahmedabu98 are the Dataflow docs up to date here? It would be a good idea for us to find a way to autogenerate some docs on supported configurations to avoid this problem FWIW (I think we've talked about this idea before)

Reply:
Yeah, Beam link is the source of truth. We should update the Dataflow link to refer to that.

Reply:
FYI. I added links to the Dataflow managed I/O pages to refer to the Beam link.